src/java/org/apache/cassandra/io/sstable/SSTable.java

```diff
@@ -101,6 +102,11 @@ protected SSTable(Descriptor descriptor, Set<Component> components, TableMetadat
         this.optimizationStrategy = Objects.requireNonNull(optimizationStrategy);
     }
 
+    public Set<Component> getComponents()
```

Comment:
Nit: this is only used by tests, we should probably mark it with `@VisibleForTesting`.